server/rest/src/main/java/org/infinispan/rest/server/authentication/BasicAuthenticator.java

```diff
@@ -0,0 +1,40 @@
+package org.infinispan.rest.server.authentication;
+
+import java.util.Base64;
+import java.util.Optional;
+
+import org.infinispan.rest.server.InfinispanRequest;
+import org.infinispan.rest.server.authentication.exceptions.AuthenticationException;
+import org.jboss.resteasy.plugins.server.embedded.SecurityDomain;
+
+public class BasicAuthenticator implements Authenticator {
+
+   private final SecurityDomain domain;
+   private final String authenticateHeader;
+
+   public BasicAuthenticator(SecurityDomain domain, String realm) {
+      this.domain = domain;
```

Comment:
Let us get rid of the worthless RestEasy SecurityDomain stuff and use CallbackHandlers instead.